rubygem/spec/load_tracking_spec.rb

```diff
@@ -30,12 +32,74 @@
 
     context '.features_loaded_by' do
       it 'returns list of new files loaded when block executes' do
-        new_files = Zeus::LoadTracking.features_loaded_by do
+        new_files, = Zeus::LoadTracking.features_loaded_by do
           $untracked_features << "an_untracked_feature.rb"
         end
 
         expect(new_files).to eq(["an_untracked_feature.rb"])
       end
     end
   end
+
+  describe '.features_loaded_by' do
+    def expect_to_load(expect_features, expect_err=NilClass)
+      new_files, err = Zeus::LoadTracking.features_loaded_by do
+        yield
+      end
+
+      expect(new_files.sort).to eq(expect_features.sort)
+      expect(err).to be_instance_of(expect_err)
+    end
+
+    def expand_asset_path(path)
+      File.join(__dir__, 'assets', path)
+    end
+
+    context 'loading valid code' do
+      it 'tracks successful require_relative' do
+        expect_to_load([expand_asset_path('require_relative.rb')]) do
+          require_relative 'assets/require_relative'
+        end
+      end
+
+      it 'tracks successful require' do
+        expect_to_load([expand_asset_path('require.rb')]) do
+          require expand_asset_path('require')
+        end
+      end
+
+      it 'tracks loads' do
+        expect_to_load([expand_asset_path('load.rb')]) do
+          load expand_asset_path('load.rb')
+        end
+      end
+    end
+
+    context 'loading invalid code' do
```

Comment:
:+1: :+1: :100: :heart: :yellow_heart: :blue_heart: this is going to be an AWESOME addition to zeus